numpy/core/numeric.py

```diff
@@ -406,7 +406,7 @@ def count_nonzero(a, axis=None):
     array([2, 3])
 
     """
-    if axis is None or axis == ():
+    if axis is None or (type(axis) == tuple and axis == ()):
```

Comment:
nitpick, but should be `type(axis) is tuple`. Though perhaps this should just be `isinstance(axis, tuple) and axis == ()`? (should we preclude, e.g., a `namedtuple`?)